cli/cmds/init.js

```diff
@@ -16,28 +16,22 @@
 
 'use strict';
 
+const colorString = require('color-string');
 const TwaGenerator = require('../../lib/TwaGenerator');
-const {jdkHelper} = require('../../lib/jdk');
-const fetch = require('node-fetch');
+const TwaManifest = require('../../lib/TwaManifest');
+const {keytool} = require('../../lib/jdk');
 const {promisify} = require('util');
-const util = require('../../lib/util');
 const colors = require('colors/safe');
 const prompt = require('prompt');
 const validUrl = require('valid-url');
+const fs = require('fs');
 prompt.get = promisify(prompt.get);
 
-// Regex for disallowed characters on Android Packages, as per
-// https://developer.android.com/guide/topics/manifest/manifest-element.html#package
-const DISALLOWED_ANDROID_PACKAGE_CHARS_REGEX = /[^ a-zA-Z0-9_\.]/;
```

Comment:
Where has all the data validation gone?

Reply:
This bit was moved to `TwaManifest.js.`. This wasn't used to validate user input, but the replace invalid characters when generating a Package ID from the host. 

There's also a `validate()` method inside `TwaManifest`, which has the code moved from `TwaGenerator`. This also needs more work, but a subject for another PR.

Reply:
Created #33 to track this.